contracts/DefaultStrategy.sol

```diff
@@ -114,23 +114,187 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
      * @param _account The address of the Account contract.
-     * @param _groupHealth The address of the Group health contract.
-     * @param _allowedStrategy The address of the AllowedStrategy contract.
+     * @param _groupHealth The address of the GroupHealth contract.
+     * @param _specificGroupStrategy The address of the SpecificGroupStrategy contract.
      */
     function setDependencies(
         address _account,
         address _groupHealth,
-        address _allowedStrategy
+        address _specificGroupStrategy
     ) external onlyOwner {
-        require(_account != address(0), "account null address");
-        require(_groupHealth != address(0), "validate group null address");
-        require(_allowedStrategy != address(0), "allowedStrategy null address");
+        require(_account != address(0), "Account null");
+        require(_groupHealth != address(0), "GroupHealth null");
+        require(_specificGroupStrategy != address(0), "SpecificGroupStrategy null");
 
-        groupHealthContract = IGroupHealth(_groupHealth);
-        allowedStrategyContract = IAllowedStrategy(_allowedStrategy);
+        groupHealth = IGroupHealth(_groupHealth);
+        specificGroupStrategy = ISpecificGroupStrategy(_specificGroupStrategy);
         account = IAccount(_account);
     }
 
+    /**
+     * @notice Distributes votes by computing the number of votes each active
+     * group should receive, then calling out to `Account.scheduleVotes`.
+     * @param votes The amount of votes to distribute.
+     * @dev The vote distribution strategy is to try and have each validator
+     * group to be receiving the same amount of votes from the system. If a
+     * group already has more votes than the average of the total available
+     * votes it will not be voted for, and instead we'll try to evenly
+     * distribute between the remaining groups.
+     * @dev Election.sol sets a dynamic limit on the number of votes receivable
+     * by a group, based on the group's size, the total amount of Locked
+     * CELO, and the total number of electable validators. We don't want to
+     * schedule votes for a group when the amount would exceed this threshold.
+     * `getVotableGroups` below selects those groups that could receive the
+     * entire `votes` amount, and filters out the rest. This is a heuristic:
+     * when distributing votes evenly, the group might receive less than
+     * `votes`, and the total amount could end up being under the limit.
+     * However, doing an exact computation would be both complex and cost a lot
+     * of additional gas, hence the heuristic. If indeed all groups are close to
+     * their voting limit, causing a larger deposit to revert with
+     * NoVotableGroups, despite there still being some room for deposits, this
+     * can be worked around by sending a few smaller deposits.
+     */
+    function generateDefaultStrategyGroupsVotesToDistributeTo(uint256 votes)
```

Comment:
Groups -> Group
For brevity, consider not referring to DefaultStrategy in this name and some others in this file - should be clear which strategy this is all for given they're in DefaultStrategy.sol.